### PR TITLE
Upgrading to JDK8

### PR DESCRIPTION
**Pull request by krmahadevan**
Fixes # .

### Did you remember to?

- [ ] Add test case(s)
- [ ] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

**Comment by krmahadevan** [comment]
This PR is a follow up to the conversation https://github.com/cbeust/testng/pull/1712#issuecomment-374099022

**Review by juherr — COMMENTED** [review]
I'm fine with it, but I'd like to take to the opportunity to remove all deprecated things in a major version.

**Comment by krmahadevan** [comment]
@juherr - Thanks for your feedback. I can raise a PR for taking care of that as well.

File: build.gradle [hunk 1]
```diff
@@ -142,7 +142,7 @@ allprojects {
         if (JavaVersion.current().isJava8Compatible()) {
```
Comment (juherr) [inline build.gradle]: I think the if can be removed

File: build.gradle [hunk 2]
```diff
@@ -57,8 +57,8 @@ apply plugin: 'osgi'
 apply plugin: 'version-injection'
 apply plugin: 'eclipse'
 
-targetCompatibility = "1.7"
-sourceCompatibility = "1.7"
+targetCompatibility = "1.8"
```
Comment (sskorol) [inline build.gradle]: it's not required if we use the same value for both source and target... according to [gradle docs](https://docs.gradle.org/current/userguide/java_plugin.html#_other_convention_properties), target = source by default.

**Review by cbeust — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (krmahadevan) [reply build.gradle]: Fixed

File: build.gradle (hunk 2 above)
Comment (krmahadevan) [reply build.gradle]: @sskorol - Thanks for calling that out. I have fixed that as well, to just include `sourceCompatibility`

**Comment by krmahadevan** [comment]
@cbeust - Please hold off on the merge. Travis CI is failing. But I can't figure out what is causing `org.testng.JarFileUtilsTest` to fail. All i see is 

```
org.testng.JarFileUtilsTest.generateTestJar FAILED
    java.lang.IllegalArgumentException at JarFileUtilsTest.java:29
```

Any idea, as to how I can get more insights on what is causing the failure ? The build works fine locally for me on JDK8 :(

**Comment by krmahadevan** [comment]
@cbeust @juherr - I have for now disabled the failing test viz `JarFileUtilsTest`. I will work on fixing it (I am not sure why it fails on Travis but passes locally for me) and send a separate PR soon. In the meantime, I don't want this upgrade to be stalled due to that.

**Comment by krmahadevan** [comment]
I have posted the question on https://developer.jboss.org/message/981362#981362

**Comment by krmahadevan** [comment]
@cbeust @juherr - I have finally figured out what the problem was. The issue was that I had missed checking in an xml file when I delivered `JarFileUtilsTest` changes. It worked fine locally because the file was there in my file system.

We had a pattern in our `.gitignore` file which was causing me to miss this particular xml file. 
I have now fixed it.

Please go ahead and help get this merged.

**Review by juherr — APPROVED** [review]
LGTM but we should remove deprecated things before the release

**Comment by krmahadevan** [comment]
@juherr 

> LGTM but we should remove deprecated things before the release

Sure. That will be a separate PR. I plan to raise a series of PR's for this so that it keeps the changeset small and facilitate easy review

ping @cbeust - Can we merge this, since @juherr has also approved this PR ?

**Comment by cbeust** [comment]
Nice catch, Krishnan :)

**Comment by krmahadevan** [comment]
Thanks cedric. I spent 3 days hunting down the cause of that silly mistake :)

**Pull request merged**